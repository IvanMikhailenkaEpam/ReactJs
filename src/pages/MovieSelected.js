import React, { Component } from 'react';
import connect from 'react-redux/es/connect/connect';
import SearchResults from '../components/search-result/SearchResults';
import SelectedFilm from '../components/navbar/seleced-film/SelectedFilm';
import Navbar from '../components/navbar/Navbar';
import SortBarInfo from '../components/sort-bar/sort-bar-info/SortBarInfo';
import SortBar from '../components/sort-bar/SortBar';
import MovieSearch from './MovieSearch';
import {
  getFilmRequest,
  searchFilmRequest,
} from '../store/movie-search/actions';

class MovieSelected extends Component {
  componentDidMount() {
    const { id, searchValue, searchBy } = this.props;
    this.props.getFilmRequest(id);
    this.props.searchFilmRequest(searchValue, searchBy);
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    const { id, searchValue, searchBy } = this.props;
    const oldId = prevProps.id;
    if (oldId !== id) {
      this.props.getFilmRequest(id);
      this.props.searchFilmRequest(searchValue, searchBy);
    }
  }


  render() {
    const { film, films } = this.props;
    if (!film || !film.genres) {
      return <MovieSearch />;
    }
    const sortBarInfo = `Films by ${film.genres[0]} genre`;

    return (
      <div>
        <Navbar>
          <SelectedFilm film={film} />
        </Navbar>
        <SortBar>
          <SortBarInfo text={sortBarInfo} />
        </SortBar>
        <SearchResults films={films} />
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  id: ownProps.match ? ownProps.match.params.id : '',
  searchValue: (state.movieSearch.film && state.movieSearch.film.genres) ? state.movieSearch.film.genres[0] : '',
  searchBy: 'genres',
  films: state.movieSearch.films,
  film: state.movieSearch.film,
});
const mapDispatchToProps = {
  searchFilmRequest,
  getFilmRequest,
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieSelected);
