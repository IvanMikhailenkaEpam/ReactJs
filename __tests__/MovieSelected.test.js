import React from 'react';
import { shallow } from 'enzyme';
import MovieSelected from "../src/pages/MovieSelected";

describe('MovieSelected', () => {
  it('renders MovieSearch correctly without params', () => {
    const component = shallow(<MovieSelected />);

    expect(component).toMatchSnapshot();
  });

  it('renders MovieSearch correctly with selected film id', () => {
    const component = shallow(<MovieSelected match={{params: {id: 1}}}/>);

    expect(component).toMatchSnapshot();
  });
});
