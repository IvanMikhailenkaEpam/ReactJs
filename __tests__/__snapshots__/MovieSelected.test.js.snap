// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSelected renders MovieSearch correctly with selected film id 1`] = `
<div>
  <Navbar>
    <SelectedFilm
      film={
        Object {
          "age": 1332,
          "description": "Some info",
          "director": "Tarantion 1",
          "duration": "122 min",
          "genre": "genre 1",
          "id": 1,
          "img": "https://m.media-amazon.com/images/M/MV5BNGNhMDIzZTUtNTBlZi00MTRlLWFjM2ItYzViMjE3YzI5MjljXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,686,1000_AL_.jpg",
          "name": "film 1",
          "rating": 4.5,
          "regards": "Oscar winning move",
        }
      }
    />
  </Navbar>
  <SortBar>
    <SortBarInfo
      text="Films by genre 1 genre"
    />
  </SortBar>
  <SearchResults
    films={
      Array [
        Object {
          "age": 2332,
          "description": "Some info",
          "director": "Tarantion 2",
          "duration": "100 min",
          "genre": "genre 1",
          "id": 2,
          "img": "https://m.media-amazon.com/images/M/MV5BMDFkYTc0MGEtZmNhMC00ZDIzLWFmNTEtODM1ZmRlYWMwMWFmXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_.jpg",
          "name": "film 2",
          "rating": 4.6,
          "regards": "",
        },
      ]
    }
  />
</div>
`;

exports[`MovieSelected renders MovieSearch correctly without params 1`] = `<MovieSearch />`;
